logstash-core/spec/logstash/compiler/compiler_spec.rb

```diff
@@ -65,19 +67,19 @@ def rand_meta
         expect(pipeline.original_source).to eq(sources.join("\n"))
       end
 
-      describe "applying protocol and id metadata" do
-        it "should apply the correct source metadata to all components" do
-          # TODO: seems to be a jruby regression we cannot currently call each on a stream
-          pipeline.get_plugin_vertices.each do |pv|
-            name_idx = pv.plugin_definition.name.split("_").last
-            source_protocol_idx = pv.source_with_metadata.protocol.split("_").last
-            source_id_idx = pv.source_with_metadata.id.split("_").last
-
-            expect(name_idx).to eq(source_protocol_idx)
-            expect(name_idx).to eq(source_id_idx)
-          end
-        end
-      end
+#       describe "applying protocol and id metadata" do
```

Comment:
what's going on here? :open_mouth:

Reply:
The motivation is that the method `LogStash::Compiler::compile_sources` before accepted an array `sources_with_metadata` and now accept a `PipelineConfig` object that contains the array. 
The test before verified that the compile generated a list of tree respecting the order of pipelines string provided in input, but the compile method before was called in production code only with an Array of a single SourceWithMetadata. So test verified a thing that could never happen in production code, and now using a single SourceWithMetadata and not a list the test has no meaning.

Reply:
If you agree with the unusefulness of this code I remove it

Reply:
okay I see ... if its not used I vote for removal.

if you really want to keep it (as its potentially to be resurrected later - which I really can not tell) than I would consider adding a comment (like you explained above) with a TODO/NOTE tag on why its commented out.